### PR TITLE
pacific: qa: add subvolume option flavors

### PR DESCRIPTION
**Pull request by mchangir**
backport tracker: https://tracker.ceph.com/issues/59202

---

backport of https://github.com/ceph/ceph/pull/45556
parent tracker: https://tracker.ceph.com/issues/54317

This PR also has cherry picked from commit 4322dcc (#49912) which avoids explicitly setting the default client mount path as '/'.

this backport was staged using ceph-backport.sh version 16.0.0.6848
find the latest version at https://github.com/ceph/ceph/blob/master/src/script/ceph-backport.sh

### CODE REVIEW

**Comment by vshankar** [comment]
jenkins test dashboard cephadm

**Comment by vshankar** [comment]
jenkins test dashboard cephadm

File: qa/tasks/cephfs/fuse_mount.py [hunk 1]
```diff
@@ -81,10 +81,14 @@ def _mount(self, mntopts, check_status):
             'ceph-fuse', "-f",
             "--admin-socket", "/var/run/ceph/$cluster-$name.$pid.asok",
         ]
+        fuse_cmd = [self._mount_bin, self.hostfs_mntpt]
```
Comment (vshankar) [inline qa/tasks/cephfs/fuse_mount.py:84]: @mchangir This is causing failures such as - https://pulpito.ceph.com/yuriw-2023-07-27_03:42:08-fs-wip-yuri10-testing-2023-07-26-0858-pacific-distro-default-smithi/7354247

PTAL

**Review by vshankar — CHANGES_REQUESTED** [review]
This change is causing failures such as

```
'FuseMount' object has no attribute '_mount_bin'
```

https://pulpito.ceph.com/yuriw-2023-07-27_03:42:08-fs-wip-yuri10-testing-2023-07-26-0858-pacific-distro-default-smithi/


<br class="Apple-interchange-newline">

**Comment by mchangir** [comment]
> This change is causing failures such as
> 
> ```
> 'FuseMount' object has no attribute '_mount_bin'
> ```
> 
> https://pulpito.ceph.com/yuriw-2023-07-27_03:42:08-fs-wip-yuri10-testing-2023-07-26-0858-pacific-distro-default-smithi/

removed offending line;
host_mntpt was appended to the mount command later in the code;
this issue was created due to manual conflict resolution during backporting process;

**Comment by vshankar** [comment]
jenkins retest this please

**Comment by vshankar** [comment]
> > This change is causing failures such as
> > ```
> > 'FuseMount' object has no attribute '_mount_bin'
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > https://pulpito.ceph.com/yuriw-2023-07-27_03:42:08-fs-wip-yuri10-testing-2023-07-26-0858-pacific-distro-default-smithi/
> 
> removed offending line; host_mntpt was appended to the mount command later in the code; this issue was created due to manual conflict resolution during backporting process;

I guess this is commit b9179f23db66f40f8bf90fea049c0a219a798bd3 that you mention. Could you explain the conflict itself in the commit message? That helps with reviewing.

**Comment by mchangir** [comment]
> > > This change is causing failures such as
> > > ```
> > > 'FuseMount' object has no attribute '_mount_bin'
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > https://pulpito.ceph.com/yuriw-2023-07-27_03:42:08-fs-wip-yuri10-testing-2023-07-26-0858-pacific-distro-default-smithi/
> > 
> > 
> > removed offending line; host_mntpt was appended to the mount command later in the code; this issue was created due to manual conflict resolution during backporting process;
> 
> I guess this is commit [b9179f2](https://github.com/ceph/ceph/commit/b9179f23db66f40f8bf90fea049c0a219a798bd3) that you mention. Could you explain the conflict itself in the commit message? That helps with reviewing.

added reason to the Conflicts section in the commit message

**Comment by ljflores** [comment]
Rados approved: https://trello.com/c/qRJogcXu/1827-wip-yuri2-testing-2023-08-16-1142-pacific

**Review by batrick — APPROVED** [review]
https://tracker.ceph.com/projects/cephfs/wiki/Pacific#2023-August-16-2

**Comment by mchangir** [comment]
> @mchangir The PR description should have an explanation for where [qa: avoid explicit set to client mountpoint as "/"](https://github.com/ceph/ceph/pull/51509/commits/cf6f49b7a36795db7ec83a3c3baa99d613c1eb55) comes from. I see that now but I shouldn't need to trawl through the PR links to learn why.

ack

**Comment by yuriw** [comment]
@mchangir pls fix and it's ready for merge
ref: https://trello.com/c/qRJogcXu

**Review by vshankar — APPROVED** [review]

**Pull request merged**